boards/arm/frdm_kw41z/pinmux.c

```diff
@@ -28,9 +28,15 @@ static int frdm_kw41z_pinmux_init(struct device *dev)
 	/* Red, green, blue LEDs. Note the red LED and accel INT1 are both
 	 * wired to PTC1.
 	 */
+#if CONFIG_PWM_0
```

Comment:
make this based on nodelabel instead.

Reply:
OK, I'm probably missing something here, but   if I drop the CONFIG_PWM_x altogether, then since the corresponding node is in the dts and has its status set as "okay" then the pimmux will always be enabled for PWM operation.  And that's not desirable. In the case of PWM0, its output pins are shared with GPIO pins that control the same RGB LED on the Freedomboard. The user should be allowed to select which one of the two methods for controlling the RGB led.

More generic speaking, just because something is in the DTS and is "okay", it doesn't mean that it should be enabled by the pinmux. More specifically, on the KW41, now SPI, I2C, ADC etc. are all enabled, since the corresponding nodes have the corresponding status set to "okay". If the user would like to build a simpler app, it'd need to modify the DTS and set the corresponding nodes to "disabled", as opposed to disabling options in the menuconfig.

So my thought is that the selection of the pinmux should be done based on the existing HW resources AND the user desired configuration.

Let me know your thoughts on this.